src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -147,15 +141,17 @@ private void validateStreamId(final List<NakadiCursor> cursors, final String str
     }
 
     public void resetCursors(final String subscriptionId, final List<NakadiCursor> cursors)
-            throws ServiceUnavailableException, NoSuchSubscriptionException, CursorUnavailableException,
+            throws ServiceUnavailableException, NoSuchSubscriptionException,
             UnableProcessException, OperationTimeoutException, ZookeeperException,
-            InternalNakadiException, NoSuchEventTypeException {
+            InternalNakadiException, NoSuchEventTypeException, InvalidCursorException {
         if (cursors.isEmpty()) {
             throw new UnableProcessException("Cursors are absent");
         }
         final Subscription subscription = subscriptionRepository.getSubscription(subscriptionId);
-        validateSubscriptionResetCursors(subscription, cursors);
-
+        validateCursorsBelongToSubscription(subscription, cursors);
+        for (final NakadiCursor cursor : cursors) {
+            cursor.validate();
```

Comment:
this method validates much less than `validateSubscriptionResetCursors` used to validate.